[quant][graphmode] Add prim::ListConstruct to general op handling

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35130 [quant][graphmode] Make interpolate/upsample work again
* #35142 [quant][graphmode] SwapDeQuant support prim::If
* #35141 [quant][graphmode][refactor] getGeneralOpTensorInputIndexes -> getGeneralOpTensorInputs
* #35135 [quant][graphmode][refactor] swapDeQuant takes block as arugment
* #35077 [quant][graphmode] Fold quantized prepacking ops
* #35073 [quant] Make conv2d_prepack and linear_prepack pure
* #34855 [quant][graphmode] quantization support for prim::CallFunction
* #34346 [quant][graphmode] Add quantization support for aten::cat
* **#34345 [quant][graphmode] Add prim::ListConstruct to general op handling**
* #34854 [quant][graphmode] quantization support for view, transpose, contiguos
* #34806 [quant][graphmode] quantization support for aten::chunk
* #34807 [quant][graphmode] quantization support for prim::ListUnpack
* #34804 [quant][graphmode] Replicate quantize node for prim::If

Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

raghuramank100 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -172,6 +172,12 @@ std::vector<size_t> getGeneralOpTensorInputIndexes(Node* n) {
                  /* call_funcs = */ {},
                  /* aten_funcs = */ single_input_aten_funcs)) {
     return {0};
+  } else if (n->kind() == prim::ListConstruct) {
```
Can you add a test that exercises this code?

jerryzh168 replied:
this will be tested in https://github.com/pytorch/pytorch/pull/34346

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat
https://github.com/pytorch/pytorch/pull/34346

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat
https://github.com/pytorch/pytorch/pull/34346

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat
https://github.com/pytorch/pytorch/pull/34346

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat
https://github.com/pytorch/pytorch/pull/34346

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat
https://github.com/pytorch/pytorch/pull/34346

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat
https://github.com/pytorch/pytorch/pull/34346

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat
https://github.com/pytorch/pytorch/pull/34346

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat
https://github.com/pytorch/pytorch/pull/34346

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat
https://github.com/pytorch/pytorch/pull/34346

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add prim::ListConstruct to general op handling"


Summary:
prim::ListConstruct is similar to an op that doesn't require observation
we want to make sure we can propagate observed property through it

Test Plan:
this will be tested when we add support for cat

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20524455](https://our.internmc.facebook.com/intern/diff/D20524455)

[ghstack-poisoned]
